sys/kern/vfs_syscalls.c

```diff
@@ -157,6 +158,36 @@ int do_fcntl(proc_t *p, int fd, int cmd, int arg, int *resp) {
       error = fd_set_cloexec(p->p_fdtable, fd, cloexec);
       break;
 
+    case F_GETFD:
+      error = fd_get_cloexec(p->p_fdtable, fd, resp);
+      break;
+
+    case F_GETFL:
```

Comment:
I'm not 100% sure if `F_GETFL` & `F_SETFL` should handle all flags. Please consult `fcntl` manual page:
* https://netbsd.gw.com/cgi-bin/man-cgi?fcntl
* https://www.freebsd.org/cgi/man.cgi?query=fcntl
* http://man7.org/linux/man-pages/man2/fcntl.2.html

Verify your findings with `sys_fcntl` NetBSD implementation: http://bxr.su/NetBSD/sys/kern/sys_descrip.c#339